crates/host/src/wasmbus/handler.rs

```diff
@@ -163,293 +161,356 @@ impl Handler {
-            res.map_err(|err| anyhow!(err).context("function failed"))?;
-        tx.await.context("failed to transmit parameters")?;
-        Ok(blobstore::container::ObjectMetadata {
-            name,
-            container: container.to_string(),
-            created_at,
-            size,
-        })
-    }
-
-    #[instrument(level = "trace", skip(self))]
-    async fn clear_container(&self, container: &str) -> anyhow::Result<()> {
-        use wrpc_interface_blobstore::Blobstore;
-
-        let wrpc = self.wrpc_blobstore_blobstore().await?;
-        let (res, tx) = wrpc
-            .invoke_clear_container(container)
-            .await
-            .context("failed to invoke `wrpc:blobstore/blobstore.clear-container`")?;
-        // TODO: return a result directly
-        res.map_err(|err| anyhow!(err).context("function failed"))?;
-        tx.await.context("failed to transmit parameters")?;
-        Ok(())
-    }
-
-    #[instrument(level = "trace", skip(self))]
-    async fn copy_object(
-        &self,
-        src_container: String,
-        src_name: String,
-        dest_container: String,
-        dest_name: String,
-    ) -> anyhow::Result<()> {
-        use wrpc_interface_blobstore::{Blobstore, ObjectId};
-
-        let wrpc = self.wrpc_blobstore_blobstore().await?;
-        let (res, tx) = wrpc
-            .invoke_copy_object(
-                &ObjectId {
-                    container: src_container,
-                    object: src_name,
-                },
-                &ObjectId {
-                    container: dest_container,
-                    object: dest_name,
-                },
-            )
-            .await
-            .context("failed to invoke `wrpc:blobstore/blobstore.copy-object`")?;
-        // TODO: return a result directly
-        res.map_err(|err| anyhow!(err).context("function failed"))?;
-        tx.await.context("failed to transmit parameters")?;
-        Ok(())
-    }
-
-    #[instrument(level = "trace", skip(self))]
-    async fn move_object(
-        &self,
-        src_container: String,
-        src_name: String,
-        dest_container: String,
-        dest_name: String,
-    ) -> anyhow::Result<()> {
-        use wrpc_interface_blobstore::{Blobstore, ObjectId};
-
-        let wrpc = self.wrpc_blobstore_blobstore().await?;
-        let (res, tx) = wrpc
-            .invoke_move_object(
-                &ObjectId {
-                    container: src_container,
-                    object: src_name,
-                },
-                &ObjectId {
-                    container: dest_container,
-                    object: dest_name,
-                },
-            )
-            .await
-            .context("failed to invoke `wrpc:blobstore/blobstore.move-object`")?;
-        // TODO: return a result directly
-        res.map_err(|err| anyhow!(err).context("function failed"))?;
-        tx.await.context("failed to transmit parameters")?;
-        Ok(())
-    }
-}
+// #[async_trait]
```

Comment:
You'll need to adapt these trait implementations using the new transport API.
We can either do it here directly, most probably using `invoke_values` https://bytecodealliance.github.io/wrpc/wrpc/transport/trait.Invoke.html#method.invoke_values (currently unstable, and will change soon, likely today)

or work on updating https://github.com/wrpc/blobstore and https://github.com/wrpc/http and integrating here

Reply:
FYI, almost ready, will have to finish next week https://github.com/bytecodealliance/wrpc/pull/147
(note the `invoke_values` and `serve_values` APIs)